src/hub.ts

```diff
@@ -18,10 +19,12 @@ export class Hub extends EventEmitter {
 
     public autoSubscribe: boolean = true;
     public useSpeedMap: boolean = true;
-    public type: Consts.HubType = Consts.HubType.UNKNOWN;
+    protected _type: Consts.HubType = Consts.HubType.UNKNOWN;
 
-    protected _ports: {[port: string]: Port} = {};
-    protected _virtualPorts: {[port: string]: Port} = {};
+    protected _attachedDevices: Device[] = [];
```

Comment:
Given the current use I would say a map of portId to Device would make more sense than a flat array for `_attachedDevices`.

Reply:
I think so too. But object literals don't accept numbers as key type, and I was hoping not to use ES6 maps. I guess we could though (_"Just shut up and do it, Nathan!"_)

Reply:
I'm totally wrong - they *do* support numbers as keys/property names, just not in initializers. I'll change to use this. :)

Reply:
Do you mean initializes as in: `({0:'Test'})[0]`? It does work for me. What's wrong with ES6 Maps, btw? I strongly prefer them to using POJOs for maps, especially in Typescript.